code/modules/recycling/disposal.dm

```diff
@@ -1138,11 +1137,10 @@
 	icon_state = "pipe-t"
 	var/obj/linked 	// the linked obj/machinery/disposal or obj/disposaloutlet
 
-/obj/structure/disposalpipe/trunk/New()
-	..()
+/obj/structure/disposalpipe/trunk/Initialize(mapload)
+	. = ..()
 	dpdir = dir
-	spawn(1)
-		getlinked()
+	addtimer(CALLBACK(src, .proc/getlinked), 0) // This has a delay of 0, but wont actually start until the MC is done
```

Comment:
This `addtimer()` should be better than the removed spawn, and more instant when running the round as opposed to a 1 tick delay to account for mapping weirdness.